components/script/dom/document.rs

```diff
@@ -249,6 +249,12 @@ pub enum DeclarativeRefresh {
     CreatedAfterLoad,
 }
 
+#[derive(JSTraceable, MallocSizeOf)]
+pub enum FontsState {
+    Unfulfilled(bool),
+    Fulfilled(Dom<FontFaceSet>),
```

Comment:
These names aren't quite right: Unfulfilled(true) actually means that the promise has been marked fulfilled. Meanwhile, we can have a Fulfilled value where the promise has not yet been fulfilled. How about `Uninitialized(bool)` and `Initialized(Dom<FontFaceSet>)`?